design-book/src/roadmap.md

```diff
@@ -0,0 +1,100 @@
+# Roadmap
+
+Making an editor is a huge project! How do we even know where to begin?
+
+Like usual, we're starting by building something small. While we won't be adopting any of the existing prototypes directly to avoid drowning in a new sprawling code-base, please look to them aggressively for inspiration on both features and implementations.
+
+The stages contained below are a flexible, coarse resolution roadmap to give newcomers and onlookers a sense of the overall plan and progress.
+If you'd like to add items to this roadmap, open a PR and ping `@alice-i-cecile` as needed for dispute resolution.
+Try to keep each stage small, simple, and easily accomplished.
+We can easily add more stages, but trying to jump ahead to a complex feature risks killing all momentum.
+For actionable, well-scoped items, please open issues instead.
+
+As you complete tasks, check them off here!
+The tasks don't need to be done *well* to be considered completed here: the Bevy community is very good at cleaning up loose strings as we work.
+
+Feel free to tackle any listed item in any order: foundations and primitives can often be built out well before the features that need them are in place.
+This list is not "all of the features that the editor will have": other stages will follow as these features are added.
+It's best not to plan too far in advance!
+
+## Stage 0: Hello Project
+
+- [x] make a repo
+- [x] add a README
+- [x] add a CONTRIBUTING.md
+- [x] add a basic Bevy project
+- [x] define the broad plan for what we're doing
+- [x] define basic constraints for how we're building the editor
+- [x] define a shared vision for what the editor is (and is not)
+
+## Stage 1: Technically an Editor
+
+- [ ] can load scenes from disk using a native file picking widget
+- [ ] can display scenes in a viewer
```

Comment:
Certainly. I'll improve the English message to make it clearer and more coherent:

The implementation of this part of the plan is unclear. Currently, Bevy doesn't support serialization of Handle<StandardMaterial> and Handle<Mesh> in a way that allows them to be loaded after the scene is loaded into the game. As a result, games typically require additional code to create material and mesh components. This poses a problem for the editor: without access to the game's spawn code, it would only be able to display a point cloud or a collection of AABB cuboids, which is far from ideal.

I see two potential solutions to this issue:

1. Allow editor_prototypes to independently create components from which meshes, materials, or entire GLTF scenes can be spawned. This would enable games that use a compatible set of components to be properly displayed in the visualizer.

2. Provide the editor with access to the game's code, allowing it to correctly initialize meshes and materials.

These approaches lead to very different editor designs and development paths. Therefore, I suggest either including this consideration in the plan or at least noting that this problem needs to be addressed.

It's worth mentioning that this issue might be more relevant to Stage 2 rather than Stage 1 of the development process. Additionally, it's possible that a solution for unified serialization of meshes and materials has already been implemented, and I've simply overlooked it.

Reply:
Your point about handles is one of the main things bsn is designed to resolve. 

Alice, it might be helpful if bsn was mentioned in this doc. "Edits bsn ast" is the fundamental design goal, as far as I am concerned, it would do to make that clearer.

Reply:
I think part 2 is relevant to what I mentioned in my comments. (And before on discord in a more general debate.)
Being able to communicate to the editor what code is relevant for spawning and rendering (and other things the editor will want to intercept) is imho key to many situations. (Scene viewing, Launching the game midway, hot-reloading custom assets, custom editors and inspectors, and more probably.)

I suppose this part needs an own design doc itself.